src/kafe_consumer_commiter.erl

```diff
@@ -0,0 +1,789 @@
+% @hidden
+-module(kafe_consumer_commiter).
+-compile([{parse_transform, lager_transform}]).
+-behaviour(gen_server).
+
+-include("../include/kafe.hrl").
+-ifdef(TEST).
+-include_lib("eunit/include/eunit.hrl").
+-endif.
+
+%% API
+-export([start_link/3]).
+
+%% gen_server callbacks
+-export([init/1
+         , handle_call/3
+         , handle_cast/2
+         , handle_info/2
+         , terminate/2
+         , code_change/3]).
+
+-define(SERVER, ?MODULE).
+
+-record(state, {
+          topic,
+          partition,
+          group_id,
+          commit_store_key,
+          commits = []
+         }).
+
+start_link(Topic, Partition, GroupID) ->
+  gen_server:start_link(?MODULE, [Topic, Partition, GroupID], []).
+
+% @hidden
+init([Topic, Partition, GroupID]) ->
+  erlang:process_flag(trap_exit, true),
+  CommitStoreKey = erlang:term_to_binary({Topic, Partition}),
+  kafe_consumer_store:insert(GroupID, {commit_pid, CommitStoreKey}, self()),
+  {ok, #state{
+          topic = Topic,
+          partition = Partition,
+          group_id = GroupID,
+          commit_store_key = CommitStoreKey,
+          commits = []
+         }}.
+
+% @hidden
+handle_call({commit, Topic, Partition, Offset, GroupID, GenerationID, MemberID, Options}, _From, State) ->
+  lager:debug("Will commit offset ~p for topic ~s, partition ~p", [Offset, Topic, Partition]),
+  case kafe_consumer_store:lookup(GroupID, allow_unordered_commit) of
+    {ok, true} ->
+      unordered_commit(Topic, Partition, Offset, GroupID, GenerationID, MemberID, Options, State);
+    _ ->
+      ordered_commit(Topic, Partition, Offset, GroupID, GenerationID, MemberID, Options, State)
+  end;
+handle_call({store_for_commit, Offset}, _From, #state{commits = Commits,
+                                                      topic = Topic,
+                                                      partition = Partition,
+                                                      group_id = GroupID} = State) ->
+  lager:debug("Store for commit offset ~p for Topic ~p, partition ~p", [Offset, Topic, Partition]),
+  GenerationID = kafe_consumer_store:value(GroupID, generation_id),
+  MemberID = kafe_consumer_store:value(GroupID, member_id),
+  CommitsList = lists:append(Commits,
+                             [{{Topic, Partition, Offset, GroupID, GenerationID, MemberID}, false}]),
+  kafe_metrics:consumer_partition_pending_commits(GroupID, Topic, Partition, length(CommitsList)),
+  {reply,
+   kafe_consumer:encode_group_commit_identifier(self(), Topic, Partition, Offset, GroupID, GenerationID, MemberID),
```

Comment:
The `kafe_consumer_commiter`'s PID is embedded here in the returned "CommitRef", for later use when requesting a commit.

However, what will happen if the `kafe_consumer_commiter` is restarted for any reason between this `store_for_commit` and the subsequent `commit` call ? The PID stored in the "CommitRef" will be stale, won't it?